src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/Constants.cs

```diff
@@ -1,14 +1,17 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using Microsoft.Extensions.PlatformAbstractions;
+
 namespace Microsoft.AspNetCore.Server.Kestrel.Infrastructure
 {
     internal class Constants
     {
         public const int ListenBacklog = 128;
 
         public const int EOF = -4095;
-        public const int ECONNRESET = -4077;
+        public static readonly int? ECONNRESET = GetECONNRESET();
```

Comment:
I ran a micro-benchmark using BenchmarkDotNet:

```
                        Method |    Median |    StdDev |
------------------------------ |---------- |---------- |
                  CompareConst | 0.0005 ns | 0.0110 ns |
         CompareStaticReadonly | 0.0000 ns | 0.0000 ns |
 CompareStaticReadonlyNullable | 0.2871 ns | 0.0015 ns |
```

Nullable is slower, but it might not matter in a real-world benchmark like Plaintext.